### PR TITLE
Unify expiry condition in getWithInfo() and onEvict()

### PR DESCRIPTION
**Pull request by chen-zhuohan**
Closes: #318

### CODE REVIEW

File: shard.go [hunk 1]
```diff
@@ -50,12 +50,11 @@ func (s *cacheShard) getWithInfo(key string, hashedKey uint64) (entry []byte, re
 	}
 
 	entry = readEntry(wrappedEntry)
-	oldestTimeStamp := readTimestampFromEntry(wrappedEntry)
-	s.lock.RUnlock()
-	s.hit(hashedKey)
-	if currentTime-oldestTimeStamp >= s.lifeWindow {
```
Comment (janisz) [inline shard.go:56]: Could you add a test for this change?

Comment (chen-zhuohan) [reply shard.go:56]: Done

**Comment by janisz** [comment]
@chen-zhuohan it looks like we have race now

**Comment by chen-zhuohan** [comment]
it seems be some problem about uint64(out of the uint64 limits).

```go
func (s *cacheShard) isExpired(oldestEntry []byte, currentTimestamp uint64) bool {
	oldestTimestamp := readTimestampFromEntry(oldestEntry)
	fmt.Println(time.Unix(int64(currentTimestamp), 0), time.Unix(int64(oldestTimestamp), 0), currentTimestamp-oldestTimestamp, s.lifeWindow, currentTimestamp-oldestTimestamp > s.lifeWindow)
	return currentTimestamp-oldestTimestamp > s.lifeWindow
}
```

I get this print:
`2022-05-30 16:02:56 +0800 CST 2022-05-30 16:02:57 +0800 CST 18446744073709551615 600 true`

see 18446744073709551615, and the problem will be clean.

I add another check to fix it.

**Review by janisz — APPROVED** [review]

**Comment by chen-zhuohan** [comment]
how long will the change merged QAQ

**Pull request merged**